### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by Diliaver**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,34 +1,79 @@
...
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
+    def __init__(self, ships, field={}):
+        self.ships = ships
+        self.field = field
+        self.field = Battleship.create(self)
+
+    def create(self):
```
Comment (IvanRamyk) [inline app/main.py:57]:
```suggestion
    def create_field(self):
```
Using the verb `create` isn't enough to describe what this method actually does

File: app/main.py [hunk 2]
```diff
@@ -1,34 +1,79 @@
...
+        list_ships = []
+        for ships in self.ships:
+            battle_ships = list(ships)
```
Comment (IvanRamyk) [inline app/main.py:60]: Why convert a ship to the list?

File: app/main.py [hunk 3]
```diff
@@ -1,34 +1,79 @@
...
+            list_ships.append(Ship(battle_ships[0], battle_ships[1]))
+        for create_decks in list_ships:
```
Comment (IvanRamyk) [inline app/main.py:62]:
```suggestion
        for ship in list_ships:
```
use nouns to name variables. it must be obvious what is stored inside

File: app/main.py [hunk 4]
```diff
@@ -1,34 +1,79 @@
...
+            create_decks.decks_create()
+        for crate_field in list_ships:
```
Comment (IvanRamyk) [inline app/main.py:64]: same issue

File: app/main.py [hunk 5]
```diff
@@ -1,34 +1,79 @@
...
+            for j in range(len(crate_field.decks) - 1):
+                self.field.update({(crate_field.decks[j].row,
```
Comment (IvanRamyk) [inline app/main.py:66]:
```suggestion
                self.field.update[(crate_field.decks[j].row, crate_field.decks[j].column] = crate_field
```
This way of creating a pair in dict is more compact

File: app/main.py [hunk 6]
```diff
@@ -1,34 +1,79 @@
...
     def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+
+    def decks_create(self):
+        list_start = list(self.start)
```
Comment (IvanRamyk) [inline app/main.py:16]: It looks like useless converting

**Review by IvanRamyk — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 5 above)
Comment (Diliaver) [reply app/main.py:66]: TypeError: 'builtin_function_or_method' object does not support item assignment

Comment (IvanRamyk) [reply app/main.py]: Sorry, forgot to delete `update` word:
```
 self.field[(crate_field.decks[j].row, crate_field.decks[j].column] = crate_field
```

**Review by IvanRamyk — APPROVED** [review]
Good job!

**Pull request closed without merging**